org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionNodeFound.java

```diff
@@ -44,4 +48,16 @@ public CompletionNodeFound(ASTNode astNode, Scope scope) {
 public CompletionNodeFound(ASTNode astNode, Scope scope, boolean insideTypeAnnotation) {
 	this(astNode, null, scope, insideTypeAnnotation);
 }
+public <T> T throwOrDeferAndReturn(Supplier<T> value) {
+	// don't yet throw this CompletionNodeFound if fields are not yet present
```

Comment:
How are the call sites for this method chosen ? Have we covered all possible cases ? I don't have a ready issue to report, but ...

Reply:
Generally, this would be needed for completion nodes within an annotation value / member value pair, which already restricts it a good deal.

I figured `CompletionOnQualifiedNameReference` would be the next candidate and just now I succeeded to create a test that actually needs that.

Now I'm running out of ideas what shape of node could appear in this context :)